src/PowerLimiter.cpp

```diff
@@ -223,7 +223,21 @@ void PowerLimiterClass::setNewPowerLimit(std::shared_ptr<InverterAbstract> inver
             _lastCommandSent = millis();
         } 
         MessageOutput.printf("[PowerLimiterClass::loop] Limit Non-Persistent: %d W\r\n", newPowerLimit);
-        inverter->sendActivePowerControlRequest(Hoymiles.getRadio(), newPowerLimit, PowerLimitControlType::AbsolutNonPersistent);
+
+        std::list<ChannelNum_t> dcChnls = inverter->Statistics()->getChannelsByType(TYPE_DC);
+        int dcProdChnls = 0, dcTotalChnls = dcChnls.size();
+        for (auto& c : dcChnls) {
+            if (inverter->Statistics()->getChannelFieldValue(TYPE_DC, c, FLD_PDC) > 0) {
+                dcProdChnls++;
+            }
+        }
+        int32_t effPowerLimit = round(newPowerLimit * static_cast<float>(dcTotalChnls) / dcProdChnls);
+        uint16_t inverterMaxPower = inverter->DevInfo()->getMaxPower();
+        if (effPowerLimit > inverterMaxPower) {
+            effPowerLimit = inverterMaxPower;
+        }
```

Comment:
`/ dcProdChnls` could lead to a division by zero error/crash if it is 0. This change should fix this.

```suggestion

        int32_t effPowerLimit;
        if (dcProdChnls > 0) {
            effPowerLimit = round(newPowerLimit * static_cast<float>(dcTotalChnls) / dcProdChnls);
            uint16_t inverterMaxPower = inverter->DevInfo()->getMaxPower();

            if (effPowerLimit > inverterMaxPower) {
                effPowerLimit = inverterMaxPower;
            }
        } else {
            effPowerLimit = config.PowerLimiter_LowerPowerLimit;
        }
```